### DISCUSSION TITLE
Filtering image vulnerabilities by CVE and image name or gobinary path using Rego

### DISCUSSION THREAD

**Discussion by elchenberg** [aquasecurity/trivy#7218] [Q&A] [answered]
### Question

Is it possible to filter/suppress image vulnerabilities by CVE and image name and gobinary path?

I have been testing this with the vulnerability `CVE-2024-21626` and the image `docker.io/k8ssandra/k8ssandra-tools:latest@sha256:96804bd4826aa46afc48b779ca4e3d8c36166d5b10e9d10f69e95b50f5591b2b`.

My policy file looks like this:

```rego
package trivy

default ignore = false

ignore {
    input.VulnerabilityID == "CVE-2024-21626"
    input.Layer.Digest == "sha256:a8bdf3764dd56f87326e3d997488e050b55d9b756fecc24e1f250c60edb0d33c"
}
```

```sh
trivy image --scanners vuln --severity CRITICAL,HIGH docker.io/k8ssandra/k8ssandra-tools:latest@sha256:96804bd4826aa46afc48b779ca4e3d8c36166d5b10e9d10f69e95b50f5591b2b --ignore-policy ignore.rego
```

This works but I would like to replace the `input.Layer.Digest` with the image name (`.ArtifactName` in the JSON output of the `trivy image` command) or the path of the gobinary (`.Results[].Target` in the JSON output of the `trivy image` command) but they do not seem to be in the `input`.

Am I missing something?

### Target

Container Image

### Scanner

Vulnerability

### Output Format

Table

### Mode

Standalone

### Operating System

Ubuntu

### Version

```bash
Version: 0.53.0
Vulnerability DB:
  Version: 2
  UpdatedAt: 2024-07-24 12:12:12.569301801 +0000 UTC
  NextUpdate: 2024-07-24 18:12:12.569301531 +0000 UTC
  DownloadedAt: 2024-07-24 13:26:20.497287603 +0000 UTC
Java DB:
  Version: 1
  UpdatedAt: 2024-06-19 01:05:01.197169156 +0000 UTC
  NextUpdate: 2024-06-22 01:05:01.197168996 +0000 UTC
  DownloadedAt: 2024-06-19 11:54:48.784724275 +0000 UTC
Check Bundle:
  Digest: sha256:ef2d9ad4fce0f933b20a662004d7e55bf200987c180e7f2cd531af631f408bb3
  DownloadedAt: 2024-07-18 11:49:06.826123765 +0000 UTC
```

**Comment by DmitriyLewen** [comment, marked as answer]
Hello @elchenberg 
Thanks for your interest to Trivy.

You can use `path` field to skip CVE - https://aquasecurity.github.io/trivy/v0.53/docs/configuration/filtering/#trivyignoreyaml

Also take a look on [VEX](https://aquasecurity.github.io/trivy/v0.53/docs/configuration/filtering/#by-vulnerability-exploitability-exchange-vex).
You can skip CVE for image using VEX.

Regards, Dmitriy

**Reply by elchenberg** [reply]
Thank you, @DmitriyLewen! I think neither the new trivyignore format, nor VEX are supported by the Trivy Operator. That is why is was asking about Rego. So I take that as a "No". :slightly_smiling_face:
